cpp/include/cudf/strings/detail/strings_children.cuh

```diff
@@ -38,6 +39,61 @@ namespace cudf {
 namespace strings {
 namespace detail {
 
+/**
+ * @brief Gather characters to create a strings column using the given string-index pair iterator
+ *
+ * @tparam IndexPairIterator iterator over type `pair<char const*,size_type>` values
+ *
+ * @param offsets The offsets for the output strings column
+ * @param chars_size The size (in bytes) of the chars data
+ * @param begin Iterator to the first string-index pair
+ * @param strings_count The number of strings
+ * @param stream CUDA stream used for device memory operations
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return An array of chars gathered from the input string-index pair iterator
+ */
+template <typename IndexPairIterator>
+rmm::device_uvector<char> make_chars_buffer(column_view const& offsets,
```

Comment:
As indicated by the function name, buffer is probably the proper type
```suggestion
rmm::device_buffer make_chars_buffer(column_view const& offsets,
```

Reply:
I prefer the typed version `rmm::device_uvector<char>`. This is consistent with other utilities as well.
We could change the function name to `make_chars_vector` if that is more palatable.

Reply:
> This is consistent with other utilities as well.

That's the info I was missing. I'm fine with `make_chars_buffer` returning vector in this case.